### ISSUE TITLE
[1.21.2->1.21] Recipe doesn't work correctly

### ISSUE THREAD

**Issue by Pantera07** [ViaVersion/ViaBackwards#912]
### '/viaversion dump' Output

https://dump.viaversion.com/8c99a5fdaef78a42718081f88c2809a787aee9a70faaf3e7b628b82242663381

### Console Error

```
none
```

### Bug Description

Recipe for Stonecutter and Crafting Table does not work correctly.

1.21.3 Server & 1.21.3 Client:
![1 21 3](https://github.com/user-attachments/assets/8680b07e-d84f-4d40-bde1-6455e83322d9)
![1 21 3-2](https://github.com/user-attachments/assets/30f00594-b990-40c8-ac9b-cb24ec8ee8a3)

1.21.3 Server & 1.21.1 Client:
(Some recipes are missing and some recipes are not clickable.)
![1 21 1](https://github.com/user-attachments/assets/2e0ca11e-ecec-4b1b-8dc8-7169b58d5b2e)
![1 21 1-2](https://github.com/user-attachments/assets/f1d2bc68-21a3-4a60-b995-60aa536fe640)

### Steps to Reproduce

1. Open 1.21.3 server.
2. Connect to 1.21.1 client.
3. Open Stonecutter and Crafting Table and look at the recipe.

### Expected Behavior

I expected it to work fine.

### Checklist

- [X] Via plugins are only running on **EITHER** the backend servers (e.g. Paper) **OR** the proxy (e.g. Velocity), **not on both**.
- [X] I have included a ViaVersion dump.
- [X] If applicable, I have included a paste (**not a screenshot**) of the error.
- [X] I have tried a build from https://ci.viaversion.com/ and the issue still persists.

**Comment by Barvalg** [comment]
![Platform: 1.21.3--13--42a1901%20%28MC%3A%201.21.3%29](https://img.shields.io/badge/Platform-1.21.3--13--42a1901%20%28MC%3A%201.21.3%29-3498db?style=flat-square) </br>![ViaVersion (5.1.2-SNAPSHOT): Even with master](https://img.shields.io/badge/ViaVersion%20(5.1.2--SNAPSHOT)-Even%20with%20master-brightgreen?style=flat-square) </br>![ViaBackwards(5.1.2-SNAPSHOT): Even with master](https://img.shields.io/badge/ViaBackwards%20(5.1.2--SNAPSHOT)-Even%20with%20master-brightgreen?style=flat-square) </br>

**Comment by shiksyt** [comment]
+1

**Comment by Noiteira** [comment]
I am also having the same bug

**Comment by Git-North** [comment]
+1

**Comment by shiksyt** [comment]
pls fix it

**Comment by octoturt** [comment]
+1 for this. recipes in the stonecutter are all mixed up and many crafting recipes either don't show up or do nothing when clicked. i don't remember which version i was running before but i updated to the latest snapshots and it persists.

Server: Paper 1.21.3-48 (latest development build at time of writing)
ViaVersion: 5.1.2 SNAPSHOT build 1035 (from Jenkins)
ViaBackwards: 5.1.2 SNAPSHOT build 528 (from Jenkins)

[Dump Results](https://dump.viaversion.com/588400abde29bd9255acc22caeeb77ef250579a34270efa481e48f16e1adc374)

**Comment by Barvalg** [comment]
![Platform: 1.21.3--48--817550c%20%28MC%3A%201.21.3%29](https://img.shields.io/badge/Platform-1.21.3--48--817550c%20%28MC%3A%201.21.3%29-3498db?style=flat-square) </br>![ViaVersion (5.1.2-SNAPSHOT): Even with master](https://img.shields.io/badge/ViaVersion%20(5.1.2--SNAPSHOT)-Even%20with%20master-brightgreen?style=flat-square) </br>![ViaBackwards(5.1.2-SNAPSHOT): Even with master](https://img.shields.io/badge/ViaBackwards%20(5.1.2--SNAPSHOT)-Even%20with%20master-brightgreen?style=flat-square) </br>

**Comment by kennytv** [comment]
I don't think this is fixable on a protocol level, if anything it'd only be in the Paper module

**Comment by JoyousCreeper1059** [comment]
why is this not fixable, @kennytv

**Comment by azhippy** [comment]
I experienced the stonecutter recipe  issue on a Fabric server, which was resolved for me by Geyser issue Geyser 2.5.0-b721, released on December 1. 
Link to Geyser issue resolved by Geyser 2.5.0-b721. https://github.com/GeyserMC/Geyser/issues/5150

**Comment by Paperboypaddy** [comment]
> I experienced the stonecutter recipe issue on a Fabric server, which was resolved for me by Geyser issue Geyser 2.5.0-b721, released on December 1. Link to Geyser issue resolved by Geyser 2.5.0-b721. [GeyserMC/Geyser#5150](https://github.com/GeyserMC/Geyser/issues/5150)

I'm on Geyser 2.5.0-b724, issue was solved for bedrock. As for java I'm on Viaversion 5.2.0 and am still having issues with stonecutter along with crafting table recipe book.

**Comment by Thatsean** [comment]
+1 1.21.3 > 1.21.1

**Comment by ThatMG393** [comment]
+1 

Server: Fabric 1.21.4 
Client: 1.21.1

**Comment by alexandre-hallaine** [comment]
+1

Server: Fabric 1.21.4
Client: 1.21.1

**Comment by Teslicek** [comment]
I have a similar issue. Folia 1.21.4 connecting with vanilla 1.21.1. 
Custom recipes are not recognized in the book menu on 1.21.1, but can be crafted.
Clients on 1.21.4 have no issues.

1.21.1
![Image](https://github.com/user-attachments/assets/b300eed6-532c-4b17-bdc5-eb42a5c48b9f)

1.21.4

![Image](https://github.com/user-attachments/assets/2b3bc3bf-f771-462f-8094-8db6a0d33bab)

Tested with Folia 1.21.4, Viaversion 5.3.1 and Viabackwards 5.3.1.

**Comment by romanromann** [comment]
+1

**Comment by Kichura** [comment]
Finally resolved.
